explorer/src/components/instruction/MangoDetails.tsx

```diff
@@ -0,0 +1,161 @@
+import { SignatureResult, TransactionInstruction } from "@solana/web3.js";
+import { useCluster } from "providers/cluster";
+import { reportError } from "utils/sentry";
+import { InstructionCard } from "./InstructionCard";
+import { AddOracleDetailsCard } from "./mango/AddOracleDetailsCard";
+import { AddPerpMarketDetailsCard } from "./mango/AddPerpMarketDetailsCard";
+import { AddSpotMarketDetailsCard } from "./mango/AddSpotMarketDetailsCard";
+import { CancelPerpOrderDetailsCard } from "./mango/CancelPerpOrderDetailsCard";
+import { CancelSpotOrderDetailsCard } from "./mango/CancelSpotOrderDetailsCard";
+import { ChangePerpMarketParamsDetailsCard } from "./mango/ChangePerpMarketParamsDetailsCard";
+import { GenericConsumeEventsDetailsCard } from "./mango/GenericConsumeEventsDetailsCard";
+import { GenericMngoAccountDetailsCard } from "./mango/GenericMngoAccountDetailsCard";
+import { GenericPerpMngoDetailsCard } from "./mango/GenericPerpMngoDetailsCard";
+import { GenericSpotMngoDetailsCard } from "./mango/GenericSpotMngoDetailsCard";
+import { PlacePerpOrderDetailsCard } from "./mango/PlacePerpOrderDetailsCard";
+import { PlaceSpotOrderDetailsCard } from "./mango/PlaceSpotOrderDetailsCard";
+import {
+  decodeAddPerpMarket,
+  decodeAddSpotMarket,
+  decodeCancelPerpOrder,
+  decodeCancelSpotOrder,
+  decodeChangePerpMarketParams,
+  decodePlacePerpOrder,
+  decodePlaceSpotOrder,
+  parseMangoInstructionTitle,
+} from "./mango/types";
+
+export function MangoDetailsCard(props: {
+  ix: TransactionInstruction;
+  index: number;
+  result: SignatureResult;
+  signature: string;
+  innerCards?: JSX.Element[];
+  childIndex?: number;
+}) {
+  const { ix, index, result, signature, innerCards, childIndex } = props;
+
+  const { url } = useCluster();
+
+  let title;
+  try {
+    title = parseMangoInstructionTitle(ix);
+
+    switch (title) {
+      case "InitMangoAccount":
+        return (
+          <GenericMngoAccountDetailsCard
+            mangoAccountKeyLocation={1}
+            title={"InitMangoAccount"}
+            {...props}
+          />
+        );
+      case "Deposit":
+        return (
+          <GenericMngoAccountDetailsCard
+            mangoAccountKeyLocation={1}
+            title={"Deposit"}
+            {...props}
+          />
+        );
+      case "Withdraw":
+        return (
+          <GenericMngoAccountDetailsCard
+            mangoAccountKeyLocation={1}
+            title={"Withdraw"}
+            {...props}
+          />
+        );
+      case "InitSpotOpenOrders":
+        return (
+          <GenericMngoAccountDetailsCard
+            mangoAccountKeyLocation={1}
+            title={"InitSpotOpenOrders"}
```

Comment:
nit: you could pass title in all these cases instead of repeating the string

Reply:
applied your suggestion in all generic cards